src/artemisa/component.js

```diff
@@ -26,23 +18,7 @@ class AbstractWithFetches extends React.Component {
 
   tryToFetch() {
     const { state, dispatch } = this.props;
-    this.getFetches().forEach(fetch => {
-      this.dispatchFetch(fetch, state, dispatch)
-    })
-  }
-
-  dispatchFetch({ storeFieldName, call, on }, state, dispatch) {
-    if (!on(this.props, state)) {
-      return;
-    }
-    const action = ({
-      type: `${ARTEMISA}_${storeFieldName}`,
-      dataApiCall: call(this.props, state)
-    })
-    const should = shouldFetch(state.artemisa[storeFieldName], action.dataApiCall.path, s => s.path)
-    if (should) {
-      dispatch(action)
-    }
+    dispatchFetches(this.props, state, dispatch, this.getFetches())
```

Comment:
yes, I agree that having too much parameters is a pain.
The object approach is good.
Nevertheless it depends on the case.
I'm not sure if this particular case applies since all params are needed :P
But maybe for the real fetch() function